Add `allowfullscreen`, `allowpaymentrequest`, `referrerpolicy` iframe attributes

thizanne opened the pull request:
https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element
https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-iframe-referrerpolicy

Note that:
- Referrerpolicy is an enumerated attribute, usually translated as a variant, except is also accepts the empty string as a valid keyword. I added a `variant_or_empty` parser that can parse this kind of attribute, parameterised by the variant to which the empty string will be mapped (given as a string, for instance `"Empty"` for the ``Empty` polyvar).
- Some other elements such as `a` and `area` should accept referrer policies, but the MDN considers this as experimental. Although I did not add this support, this should not be too hard now that the attribute exists.

Thanks @Drup for the help.

thizanne commented:
See #240

Drup commented:
Nice work! The addition of `variant_or_empty` is interesting and I'm happy with your solution. It does mean that `referrerpolicy=empty` will be accepted in the ppx, but I don't feel terribly bad about that.

Nevertheless, I would like a ppx test for the other cases, notably to check that `things-with-dashes` are properly translated. A test with `no-referrer-when-downgrade` would do nicely.

You also need to rebase to fix the conflict in the changelog.

thizanne commented:
Sorry for the force-push noise, I apparently rebased against the wrong branch. I added the ppx test and rebased nicely (now).

As for the empty thing, I agree that's not ideal, but we won't have a nice typing solution to that: we would have to use an option type or similar, and I feel like this does not semantically reflect the fact that `""` is considered as a valid keyword, not the absence of a keyword. However this could possibly appear in the documentation -- unless we prefer to hide it under the carpet (or not to actually care since it's probably no big deal).

Drup commented:
Well, there is also the solution to write the ppx thingy by hand. But honestly, it doesn't matter. I'll merge when CI is green.
